src/compiler/checker.ts

```diff
@@ -14609,7 +14609,7 @@ namespace ts {
 
         function checkTypeOfExpression(node: TypeOfExpression): Type {
             checkExpression(node.expression);
-            return stringType;
+            return getUnionType(arrayFromMap(typeofEQFacts.keys(), s => getLiteralTypeForText(TypeFlags.StringLiteral, s)));

```

Comment:
Take a look at the new commit; I added a new `typeofType` that is created when the checker is.

Note that it changes the print order of string literal unions slightly. I think this is because 'string', 'symbol', etc are added to a union very early in the creation of the checker.